src/avx512bw/implementation.cpp

```diff
@@ -23,7 +32,30 @@ simdutf_warn_unused size_t implementation::convert_utf8_to_utf16(const char* buf
 }
 
 simdutf_warn_unused size_t implementation::convert_valid_utf8_to_utf16(const char* buf, size_t len, char16_t* utf16_output) const noexcept {
-   return scalar::utf8_to_utf16::convert_valid(buf, len, utf16_output);
+  utf8_to_utf16_result ret = avx512bw::valid_utf8_to_fixed_length<char16_t>(buf, len, utf16_output);
+  size_t saved_bytes = ret.second - utf16_output;
+  const char* end = buf + len;
+  if (ret.first == end) {
+    return saved_bytes;
+  }
+
+  // Note: AVX512 procedure looks up 4 bytes forward, and
+  //       correctly converts multi-byte chars even if their
+  //       continuation bytes lie outsiede 16-byte window.
+  //       It meas, we have to skip continuation bytes from
+  //       the beginning ret.first, as they were already consumed.
+  while (ret.first != end and ((uint8_t(*ret.first) & 0xc0) == 0x80)) {
```

Comment:
'and' -> I like it but many users consider it a bug in the programming language. It might be best not to use it... otherwise we will get reports...

Reply:
I guess some poor compilers did not catch up yet. Changed.